src/libraries/System.Private.Xml/src/System/Xml/Dom/XmlProcessingInstruction.cs

```diff
@@ -14,94 +14,93 @@ public class XmlProcessingInstruction : XmlLinkedNode
         {
-            _target = target;
-            _data = data;
-        }
+            Debug.Assert(target != null, "All callsites should have already checked for null");
 
-        // Gets the name of the node.
-        public override string Name
-        {
-            get
+            if (string.IsNullOrEmpty(target))
             {
-                if (_target != null)
-                    return _target;
-                return string.Empty;
+                throw new ArgumentException(SR.Xml_EmptyName);
             }
-        }
 
-        // Gets the name of the current node without the namespace prefix.
-        public override string LocalName
-        {
-            get { return Name; }
+            _target = target;
+            _data = data ?? string.Empty;
         }
 
-        // Gets or sets the value of the node.
+        /// <inheritdoc />
+        public override string Name => _target;
+
+        /// <inheritdoc />
+        public override string LocalName => Name;
+
+        /// <inheritdoc />
         [AllowNull]
         public override string Value
         {
-            get { return _data; }
-            set { Data = value!; } //use Data instead of data so that event will be fired
+            get => _data;
+            set { Data = value; } // use Data instead of data so that event will be fired
         }
 
         // Gets the target of the processing instruction.
-        public string? Target
-        {
-            get { return _target; }
-        }
+        public string Target => _target;
 
         // Gets or sets the content of processing instruction,
         // excluding the target.
+        [AllowNull]
         public string Data
         {
-            get { return _data; }
+            get => _data;
             set
             {
                 XmlNode? parent = ParentNode;
-                XmlNodeChangedEventArgs? args = GetEventArgs(this, parent, parent, _data, value, XmlNodeChangedAction.Change);
+                string val = value ?? string.Empty;
+                XmlNodeChangedEventArgs? args = GetEventArgs(this, parent, parent, _data, val, XmlNodeChangedAction.Change);
+
                 if (args != null)
+                {
                     BeforeEvent(args);
-                _data = value;
+                }
+
+                _data = val;
+
                 if (args != null)
+                {
                     AfterEvent(args);
+                }
             }
         }
 
-        // Gets or sets the concatenated values of the node and
-        // all its children.
+        /// <inheritdoc />
+        [AllowNull]
         public override string InnerText
         {
-            get { return _data; }
-            set { Data = value; } //use Data instead of data so that change event will be fired
+            get => _data;
+            set { Data = value; } // use Data instead of data so that change event will be fired
```

Comment:
Regarding the comment, `Data` also normalizes null to empty, whereas just setting _data = value would not.

Reply:
I updated the comment